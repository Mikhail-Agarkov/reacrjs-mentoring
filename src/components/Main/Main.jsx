import React from 'react';
import StatusBar from "../StatusBar/StatusBar";
import FilmList from "../FilmList/FilmList";
import {bindActionCreators} from 'redux';
import {connect} from "react-redux";
import {fetchMovies, searchMovies, setSortBy} from "../../redux/actions";
import {withRouter} from 'react-router-dom';
import {parse} from 'query-string';
import Filter from "../Filter/Filter";

class Main extends React.PureComponent {

    componentWillMount() {
        this.performSearch();
    }

    componentDidUpdate() {
        this.performSearch();
    }

    performSearch = () => {
        const {location} = this.props;
        if (location.pathname === '/search') {
            const values = parse(location.search);
            this.props.searchMovies(values.q, values.by, values.sort);
        } else {
            this.props.fetchMovies();
        }
    };

    render() {
        return (
            <main>
                <StatusBar>
                    <Filter title='Sort by' selected={this.props.sortBy} buttons={
                        [
                            {
                                text: 'Release date',
                                value: 'release_date',
                                handler: () => this.props.setSortBy('release_date'),
                                key: 1
                            },
                            {
                                text: 'Rating',
                                value: 'vote_average',
                                handler: () => this.props.setSortBy('vote_average'),
                                key: 2
                            }
                        ]
                    }/>
                </StatusBar>
                <FilmList/>
            </main>
        );
    }
}

const mapStateToProps = (state) => ({
    sortBy: state.movies.sortBy
});

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchMovies, searchMovies, setSortBy
}, dispatch);

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Main));
