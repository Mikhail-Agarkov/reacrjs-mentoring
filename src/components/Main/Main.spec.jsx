import React from 'react';
import {shallow} from 'enzyme';

import Main from './Main';

describe('Main', () => {
    it('should render logotype of the site correctly', () => {
        const component = shallow(<Main/>);
        expect(component).toMatchSnapshot();
    })
});