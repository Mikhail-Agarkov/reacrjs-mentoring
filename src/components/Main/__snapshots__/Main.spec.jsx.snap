// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render logotype of the site correctly 1`] = `
<main>
  <StatusBar />
  <FilmList
    films={
      Array [
        Object {
          "alt": "Final Fantasy: Kingsglaive cover",
          "date": 2016,
          "genre": "Drama/Fantasy",
          "key": 1,
          "src": "./assets/sample/ffkingsglaive.jpg",
          "title": "Final Fantasy: Kingsglaive",
        },
        Object {
          "alt": "Final Fantasy: Advent Children cover",
          "date": 2005,
          "genre": "Drama/Fantasy",
          "key": 2,
          "src": "./assets/sample/ffadventchildren.png",
          "title": "Final Fantasy: Advent Children",
        },
        Object {
          "alt": "Green Book cover",
          "date": 2018,
          "genre": "Drama/Comedy",
          "key": 3,
          "src": "./assets/sample/greenbook.jpg",
          "title": "Green Book",
        },
        Object {
          "alt": "Parasite cover",
          "date": 2019,
          "genre": "Drama/Mystery",
          "key": 4,
          "src": "./assets/sample/parasite.jpeg",
          "title": "Parasite",
        },
        Object {
          "alt": "The House That Jack Built cover",
          "date": 2018,
          "genre": "Drama/Thriller",
          "key": 5,
          "src": "./assets/sample/jackhouse.jpg",
          "title": "The House That Jack Built",
        },
        Object {
          "alt": "Joker cover",
          "date": 2019,
          "genre": "Drama/Thriller",
          "key": 6,
          "src": "./assets/sample/joker.jpg",
          "title": "Joker",
        },
      ]
    }
  />
</main>
`;
